code/mob/living/life/canmove.dm

```diff
@@ -9,9 +9,10 @@
 			owner.lying = istype(owner.buckled, /obj/stool/bed) || istype(owner.buckled, /obj/machinery/conveyor)
 			if (owner.lying)
 				owner.drop_item()
-			owner.set_density(initial(owner.density))
+			if(!(owner.mob_flags & NO_DENSITY))
+				owner.set_density(initial(owner.density))
```

Comment:
Not great, but I'd rather avoid refactoring all the life processes and things depending on it.